builder/builder_test.go

```diff
@@ -307,3 +339,175 @@ func getAppHashFromEVM(ethState *state.StateDB, header *monomer.Header) (common.
+	verifyBlockContent(t, wantBlock, gotBlock, builtBlock)
+
+	// We expect two transactions: one combined transaction (l1InfoTx + depositTxs) and one withdrawal transaction (withdrawalTx).
+	require.Equal(t, 2, builtBlock.Txs.Len(), "Expected the built block to contain 2 transactions: depositTxs and withdrawalTx")
+	require.Equal(t, 2, gotBlock.Txs.Len(), "Expected the built block to contain 2 transactions: depositTxs and withdrawalTx")
+
+	// Test parseWithdrawalMessages
+	testParseWithdrawalMessages(t, withdrawalTxResult)
+
+	expectedStateRoot := wantBlock.Header.StateRoot
+	gotStateRoot := gotBlock.Header.StateRoot
+	require.Equal(t, expectedStateRoot, gotStateRoot, "Expected the built block to contain state root hash")
+
+	for i := range gotBlock.Txs {
+		select {
+		case event, ok := <-subscription.Out():
+			if !ok {
+				require.FailNow(t, "event channel closed unexpectedly")
+			}
+			data := event.Data()
+			require.IsType(t, bfttypes.EventDataTx{}, data)
+			e, ok := data.(bfttypes.EventDataTx)
+			require.True(t, ok)
+			// Deposit transaction
+			if i == 0 {
+				require.Equal(t, e.GetResult().Events[0].Attributes[0].Value, "/rollup.v1.MsgApplyL1Txs")
+			}
+			// Withdrawal transaction
+			if i == 1 {
+				require.Equal(t, e.GetResult().Events[0].Attributes[0].Value, "/rollup.v1.MsgInitiateWithdrawal")
+			}
+
+		case <-subscription.Canceled():
+			require.FailNow(t, "subscription channel closed unexpectedly")
+		}
+		require.NoError(t, subscription.Err())
+	}
+}
+
+func testParseWithdrawalMessages(t *testing.T, withdrawalTxResult *abcitypes.TxResult) {
+	const (
+		nonceKey          = "nonce"
+		expectedEventType = "withdrawal_initiated"
+	)
+
+	var withdrawalEvent *abcitypes.Event
+	for i := range withdrawalTxResult.Result.Events {
+		event := &withdrawalTxResult.Result.Events[i]
+		if event.Type == expectedEventType {
+			withdrawalEvent = event
+			break
+		}
+	}
+	require.NotNil(t, withdrawalEvent, "Expected to find a withdrawal_initiated event")
+	// withdrawalEvent is guaranteed to be non-nil at this point due to the require.NotNil check above
+	require.NotEmpty(t, withdrawalEvent.Attributes, "Expected attributes to not be empty for withdrawal event")
+
+	var nonceAttribute *abcitypes.EventAttribute
+	for i := range withdrawalEvent.Attributes {
+		attribute := &withdrawalEvent.Attributes[i]
+		if attribute.Key == nonceKey {
+			nonceAttribute = attribute
+			break
+		}
+	}
+	require.NotNil(t, nonceAttribute, "Expected to find a withdrawal nonce attribute")
+	// nonceAttribute is guaranteed to be non-nil at this point due to the require.NotNil check above
+	require.NotEmpty(t, nonceAttribute.Value, "Withdrawal nonce value should not be empty")
+}
+
+func checkDepositTXResult(t *testing.T, txStore txstore.TxStore, depositTx bfttypes.Tx, expectedDepositAmount, cosmAddr string) {
+	const (
+		eventType              = "coin_received"
+		receiverAttributeIndex = 0
+		valueAttributeIndex    = 1
+		targetEventCount       = 2
+	)
```

Comment:
Instead of using the bank modules `coin_received` event, could we use the `x/rollup` module's `EventTypeMintETH` event and inline it instead of creating a new constant here?